### ISSUE TITLE
endless loop when two methods in different states have the same name

### ISSUE THREAD

**Issue by hrigu** [pluginaweek/state_machine#140]
given: a state machine with two states. Each state defines a method with the same name.
When I call the method of the second state
Then the program ends in an endless loop

``` ruby
require 'rspec'
require 'state_machine'

class StateWithBehavior
  state_machine initial: :idle do
    event :go do
      transition :idle => :running
      transition :running => :idle
    end
    state :idle do
      def go
        puts "start"
        super
      end
    end
    state :running do
      def go
        puts "stop"
        super
      end
    end
  end
end

describe StateWithBehavior do
  subject { StateWithBehavior.new }
  it "has idle as initial state" do
    subject.state_name.should == :idle
    subject.go  #<-works fine
    subject.state_name.should == :running
    subject.go #<-endless loop
    subject.state_name.should == :idle
  end
end
```

**Comment by hrigu** [comment]
I'm using state_machine version 1.0.2

**Comment by obrie** [comment]
Thanks for reporting this issue!  I was able to reproduce it, using Ruby 1.9.1.  The behavior appears significantly different in Ruby 1.8.7.  I'll take a look at this and let you know what I find.

**Comment by obrie** [comment]
It's worth noting that this still only works on Ruby 1.9.1+.  However, it should no longer infinitely loop :)

This change will get included in the 1.1.0 release coming sometime in the next few weeks.
